### PR TITLE
Renaming of onnx-mlir libraries

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
Some time ago, we renamed our libraries to start with OM prefix, but many of the newly added libs do not follow this pattern. I changed all the ones we use regularly. This PR renames them.

Also, the name of the compile object in the python interface used "Py" as prefix, but really should use "OM", so that was changed too.

There are many libs that are still not using the OM names, notably the NNPA and the HLO ones. I suggest that we adopt the same standard as MLIR (which prefix theirs with MLIR) and also change the prefixes of these.

The ones that are touchy are the "runtime" libs as their names may be already used explicitly, so these we should consider a bit more carefully.

No functionality was changed.

### CODE REVIEW

**Comment by AlexandreEichenberger** [comment]
@tungld could I also rename the NNPA prefixed libs, or do you prefer have the accel as prefix only.

**Comment by tungld** [comment]
> @tungld could I also rename the NNPA prefixed libs, or do you prefer have the accel as prefix only.

Yes, please do the same for NNPA. I am thinking of using OMNNPA prefix, but please feel free to make your own decision.

**Comment by AlexandreEichenberger** [comment]
@tungld Added OMNNPA prefix as requested

File: src/Runtime/CMakeLists.txt
Comment (tungld) [inline src/Runtime/CMakeLists.txt:115]: I know another place using ExecutionSession that is https://github.com/onnx/onnx-mlir/blob/main/utils/RunONNXModel.py#L121. RunONNXModel.py is used by Jenkins to check the onnx model zoo.

**Review by tungld — APPROVED** [review]
LGTM! 

Please make sure to make a change in `utils/RunONNXModel.py` before merging. Otherwise, Jenkins check for the onnx model zoo would fail.

File: src/Runtime/CMakeLists.txt
Comment (AlexandreEichenberger) [reply src/Runtime/CMakeLists.txt:115]: @tungld 

I have not renamed the module within the python interface because others may use it... I have tested this example

``` python
import numpy as np
from PyRuntime import ExecutionSession

# Load the model mnist.so compiled with onnx-mlir.
model = './mnist.so'

# Load the model mnist.so compiled with onnx-mlir.
model = './mnist.so'
session = ExecutionSession(model)
```

which is from mnist_runPyRuntime.py

Is that what the issue is?

Now if you agree that we should also change the module name inside python, to match the compiler interface...

Here is what I have (and like) for compiler module
``` python
import numpy as np
from PyCompileAndRuntime import OMCompileExecutionSession

# Load onnx model and create CompileExecutionSession object.
inputFileName = './mnist.onnx'
# Set the full name of compiled model
sharedLibPath = './mnist.so'
# Set the compile option as "-O3"
session = OMCompileExecutionSession(inputFileName,sharedLibPath,"-O3")

```

Comment (AlexandreEichenberger) [reply src/Runtime/CMakeLists.txt:115]: @tungld let me know, will not merge before I get your input on this

Comment (tungld) [reply src/Runtime/CMakeLists.txt:115]: OK, so `ExecutionSession ` here is not the one in `from PyRuntime import ExecutionSession` . Then  it's OK to merge now. Thanks!

Comment (tungld) [reply src/Runtime/CMakeLists.txt:115]:
```
Here is what I have (and like) for compiler module

import numpy as np
from PyCompileAndRuntime import OMCompileExecutionSession

# Load onnx model and create CompileExecutionSession object.
inputFileName = './mnist.onnx'
# Set the full name of compiled model
sharedLibPath = './mnist.so'
# Set the compile option as "-O3"
session = OMCompileExecutionSession(inputFileName,sharedLibPath,"-O3")
```

I see we can change  the python module from `ExecutionSession` to `OMExecutionSession` to make it consistent with `OMCompileExecutionSession`. Totally agree.

Comment (AlexandreEichenberger) [reply src/Runtime/CMakeLists.txt:115]: Renamed ExecutionSession to OMExecutionSession in the python object. Believe that I made the changes everywhere.

Comment (tungld) [reply src/Runtime/CMakeLists.txt:115]: Thanks so much! LGTM!

**Comment by tungld** [comment]
@AlexandreEichenberger could you resolve conflicts and try to merge this as soon as possible to avoid other conflicts? thanks!

**Pull request merged**
